src/Evolution/Systems/ScalarWave/BoundaryCorrections/UpwindPenalty.hpp

```diff
@@ -0,0 +1,288 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <memory>
+#include <optional>
+
+#include "DataStructures/DataBox/Prefixes.hpp"
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Evolution/Systems/ScalarWave/BoundaryCorrections/BoundaryCorrection.hpp"
+#include "Evolution/Systems/ScalarWave/Tags.hpp"
+#include "NumericalAlgorithms/DiscontinuousGalerkin/Formulation.hpp"
+#include "Options/Options.hpp"
+#include "Parallel/CharmPupable.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+class DataVector;
+namespace gsl {
+template <typename T>
+class not_null;
+}  // namespace gsl
+namespace PUP {
+class er;
+}  // namespace PUP
+/// \endcond
+
+namespace ScalarWave::BoundaryCorrections {
+/*!
+ * \brief Computes the scalar wave upwind multipenalty boundary
+ * correction.
+ *
+ * This implements the upwind multipenalty boundary correction term
+ * \f$D_\alpha\f$. The general form is given by:
+ *
+ * \f{align*}{
+ *   \label{eq:pnpm upwind boundary term characteristics}
+ *   D_\beta =
+ *   T_{\beta\hat{\beta}}^{\mathrm{ext}}
+ *   \Lambda^{\mathrm{ext},-}_{\hat{\beta}\hat{\alpha}}
+ *   v^{\mathrm{ext}}_{\hat{\alpha}}
+ *   -T_{\beta\hat{\beta}}^{\mathrm{int}}
+ *   \Lambda^{\mathrm{int},-}_{\hat{\beta}\hat{\alpha}}
+ *   v^{\mathrm{int}}_{\hat{\alpha}},
+ * \f}
+ *
+ * We denote the evolved fields by \f$u_{\alpha}\f$, the characteristic fields
+ * by \f$v_{\hat{\alpha}}\f$, and implicitly sum over reapeated indices.
+ * \f$T_{\alpha\hat{\alpha}}\f$ transforms characteristic fields to evolved
+ * fields, while \f$\Lambda_{\hat{\alpha}\hat{\beta}}^-\f$ is a diagonal matrix
```

Comment:
[minor nitpick] In lines 35-51, you often use the opposite subscript alpha/beta when talking about the terms vs in the displayed equation. Can you do some alpha <-> beta swaps either in the displayed equation or in the sentences? It's just notation, but might as well make it easier to read...

Reply:
Sorry, you would like me to use `u_beta` and `v_alpha` so it matches the equation?

Reply:
I'm suggesting either (note: i'll write (a,b) instead of (alpha,beta) for simplicity here),
1) you match the text to the equation by changing D_a -> D_b on line 35, then T_aa -> T_bb and L_ab -> L_ba in this paragraph, OR
2) match the equation to the text by changing D_b -> D_a, T_bb -> T_aa, L_ba -> L_ab, v_a -> v_b in the equation (and then also use u_b and v_b in this paragraph)

I guess it depends if you prefer to have the alpha subscript on D or on (u and v).